src/core/transaction.c

```diff
@@ -1159,6 +1164,11 @@ static bool shall_stop_on_isolate(Transaction *tr, Unit *u) {
         if (u->ignore_on_isolate)
                 return false;
 
+        /* When isolating on soft reboot, don't stop surviving units */
+        if (IN_SET(manager_state(u->manager), MANAGER_INITIALIZING, MANAGER_STARTING, MANAGER_STOPPING) &&
+                   u->survive)
+                return false;
```

Comment:
the same check again? still not getting it

(if it makes sense – which i don't see – should be hidden in a helper `unit_ignore_on_isolate()` which checks `u->ignore_on_isolate` first, and then does the survive check too)

Reply:
added the helper